src/Features/CSharp/Portable/Completion/CompletionProviders/AwaitCompletionProvider.cs

```diff
@@ -111,5 +48,28 @@ private static int GetSpanStart(SyntaxNode declaration)
                 _ => throw ExceptionUtilities.UnexpectedValue(declaration.Kind())
             };
         }
+
+        private protected override bool ShouldMakeContainerAsync(SyntaxToken token)
+        {
+            var declaration = GetAsyncSupportingDeclaration(token);
+            return declaration is not null && !declaration.GetModifiers().Any(SyntaxKind.AsyncKeyword);
```

Comment:
i wonder if this line coudl be pulled up to the base, using SYntaxGenerator.GetModifiers to unify the code.